wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java

```diff
@@ -263,6 +263,35 @@ public NodeList getFileData(String logicalname, Long beginrow, Integer numrows,
 
     }
 
+    private void checkSuperfileLayouts(String[] subfiles, String cluster) throws Exception
+    {
+        String eclrecord = null;
+        String basefile = null;
+        if (subfiles == null)
+        {
+            return;
+        }
+        for (int i = 0; i < subfiles.length; i++)
+        {
+            DFUInfoResponse details = this.getFileInfo(subfiles[i], cluster);
+            if (details == null || details.getFileDetail() == null)
+            {
+                continue;
+            }
+            String thisecl = details.getFileDetail().getEcl() == null ? "" : details.getFileDetail().getEcl();
+            if (eclrecord == null)
```

Comment:
if (eclrecord == null && thisecl.len != 0 )  ?

Reply:
no, this should only be populated once by the first record in the list. If some file's added to a superfile with an empty record structure, we still want to check that against other records.